src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/OffenderInternetAddressRepository.kt

```diff
@@ -0,0 +1,11 @@
+package uk.gov.justice.hmpps.prison.repository.jpa.repository
+
+import org.springframework.data.jpa.repository.Query
+import org.springframework.data.repository.CrudRepository
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderInternetAddress
+import java.util.Optional
+
+interface OffenderInternetAddressRepository : CrudRepository<OffenderInternetAddress, Long> {
+  @Query("Select offInternetAddress from OffenderInternetAddress offInternetAddress where offInternetAddress.offender.rootOffender.nomsId = :nomsId and offInternetAddress.internetAddressId = :internetAddressId")
+  fun findByRootNomsIdAndInternetAddressId(nomsId: String, internetAddressId: Long?): Optional<OffenderInternetAddress>
```

Comment:
I think we will need to make sure we have a pessimistic locking timeout applied to this, see recent thread on this: https://mojdt.slack.com/archives/C0767V49AHM/p1749027303287899